app/admin/books_admin.rb

```diff
@@ -5,14 +5,12 @@
 
   # Customize the table columns shown on the index view.
   #
-  # table do
-  #   column :id
-  #   column :title
-  #   column :price
-  #   column @subject.name
-  #   column :created_at, align: :center
-  #   actions
-  # end
+  table do
+    column :id
+    column :title
+    column :subject_id, -> (book) { Subject.find(book.subject_id).name }
+    column :publisher_id, -> (book) { Publisher.find(book.publisher_id).name }
```

Comment:
Both of these queries can be improved by relying on ActiveRecord a bit more. Whenever you are in a model and you say `belongs_to :subject`, that means Rails gives you a way to directly access that subject without having to explicitly query for it. You can just say `book.subject.name` and behind the scenes Rails will query to find the subject for you! Similarly for Publisher!